src/contracts/middleware/StakeRegistry.sol

```diff
@@ -0,0 +1,492 @@
+        return (
+            // check that the update specified by `stakeHistoryIndex` occurred at or prior to `blockNumber`
+            (operatorStakeUpdate.updateBlockNumber <= blockNumber)
+            &&
+            // if there is a next update, then check that the next update occurred strictly after `blockNumber`
+            (operatorStakeUpdate.nextUpdateBlockNumber == 0 || operatorStakeUpdate.nextUpdateBlockNumber > blockNumber)
+            &&
+            /// verify that the stake was zero at the time (note: here was use the assumption that the operator was 'inactive'
+            /// once their stake fell to zero)
+            operatorStakeUpdate.stake == 0
+        );
+    }
+
+    // MUTATING FUNCTIONS
+
+    /// @notice Adjusts the `minimumStakeFirstQuorum` -- i.e. the node stake (weight) requirement for inclusion in the 1st quorum.
+    function setMinimumStakeForQuorum(uint8 quorumNumber, uint96 minimumStake) external onlyServiceManagerOwner {
+        _setMinimumStakeForQuorum(quorumNumber, minimumStake);
+    }
+
+    /**
+     * @notice Registers the `operator` with `operatorId` for the specified `quorumNumbers`.
+     * @param operator The address of the operator to register.
+     * @param operatorId The id of the operator to register.
+     * @param quorumNumbers The quorum numbers the operator is registering for, where each byte is an 8 bit integer quorumNumber.
+     * @dev access restricted to the RegistryCoordinator
+     * @dev Preconditions (these are assumed, not validated in this contract):
+     *         1) `quorumNumbers` has no duplicates
+     *         2) `quorumNumbers.length` != 0
+     *         3) `quorumNumbers` is ordered in ascending order
+     *         4) the operator is not already registered
+     */
+    function registerOperator(address operator, bytes32 operatorId, bytes calldata quorumNumbers) external virtual {
+        _registerOperator(operator, operatorId, quorumNumbers);
+    }
+
+    /**
+     * @notice Deregisters the operator with `operatorId` for the specified `quorumNumbers`.
+     * @param operatorId The id of the operator to deregister.
+     * @param quorumNumbers The quorum numbers the operator is deregistering from, where each byte is an 8 bit integer quorumNumber.
+     * @dev access restricted to the RegistryCoordinator
+     * @dev Preconditions (these are assumed, not validated in this contract):
+     *         1) `quorumNumbers` has no duplicates
+     *         2) `quorumNumbers.length` != 0
+     *         3) `quorumNumbers` is ordered in ascending order
+     *         4) the operator is not already deregistered
+     *         5) `quorumNumbers` is the same as the parameter use when registering
+     */
+    function deregisterOperator(bytes32 operatorId, bytes calldata quorumNumbers) external virtual {
+        _deregisterOperator(operatorId, quorumNumbers);
```

Comment:
wait, are these functions supposed to just be unpermissioned? seems like an oversight (+ something that should have a unit test!)

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/8280cc553d1cf02df62cea8278015eed5ffd75d9

Reply:
I actually found this on my unit-test-stake-registry branch :)